book/src/dev/rfcs/0005-state-updates.md

```diff
@@ -777,14 +777,17 @@ if the block is not in any non-finalized chain:
     the `block_by_height` tree (to get the block data).
 
 
-### TODO document request for utxo is available
+### `Request::AwaitUtxo(OutPoint)`
 
-utxo_is_available:
-* is it in spent? Return false
-* is it in created? Return true
-* is it in the permanent state? Return true
-* otherwise, Return false
+Returns
+
+- `Response::Utxo(transparent::Output)`
+
+- (non-finalized) if any `Chains` contain an `OutPoint` in their `created_utxos` and not their `spent_utxo` get the `transparent::Output` from `OutPoint`'s transaction
+- (finalized) else if `OutPoint` is in `utxos_by_outpoint` return the associated `transparent::Output`.
```

Comment:
My original `utxo_is_available` was meant for contextual verification, not prospective verification. Since a chain can fork at any non-finalized block, I don't think we should check `spent_utxo` here.

Instead, we should check `spent_utxo` using the original `utxo_is_available` logic during contextual verification.

```suggestion
- (non-finalized) if any `Chains` contain `OutPoint` in their `created_utxos`, get the `transparent::Output` from `OutPoint`'s transaction
- (finalized) else if `OutPoint` is in `utxos_by_outpoint` return the associated `transparent::Output`.
```

Reply:
applied this in the test PR to avoid needing a new test and approval